File: Python PowerPoint Examples/ShapesApiDemo/addSlideNewTextRange.py
import spirecloudpowerpoint
from spirecloudpowerpoint.configuration import *

appId = "your id"
appKey = "your key"
baseUrl = "https://api.e-iceblue.cn"
configuration = Configuration(appId, appKey,baseUrl)
api= spirecloudpowerpoint.api.shapes_api.ShapesApi(configuration)

fileName = "AddSlideNewTextRange.pptx"
password = ""
storage = ""
inputFolder = "input"

slide_index = 0
shape_index = "0"
paragraph_index = 0
source = spirecloudpowerpoint.models.Portion()
source.text = "add new Text"
source.font_bold = True
source.font_color = "blue"
source.spacing = 5
result =api.add_slide_new_text_range(fileName, slide_index, shape_index, paragraph_index, source=source, password=password, storage=storage,folder=inputFolder)

File: JavaScript PowerPoint Examples/ShapesApiDemo/addSlideNewTextRange.js
(function (){
    var appId = "your ID";
    var appKey = "your Key";
    var baseUrl = "https://api.e-iceblue.cn";

    var SpirecloudPowerpoint = require('../../src/index');
    var apiClient = new SpirecloudPowerpoint.ApiClient(appId, appKey,baseUrl);
    var instance = new SpirecloudPowerpoint.ShapesApi(apiClient);
   
    var name = 'AddSlideNewTextRange.pptx';
    var folder = 'input';
    var slideIndex = 0;
    var shapeIndex = 0;
    var paragraphIndex = 0;
    var opts = {};
    opts.source = new SpirecloudPowerpoint.Portion();
    opts.source.Text = "add new Text";
    opts.source.FontBold = true;
    opts.source.Spacing = 5.0;
    opts.source.FontColor = "Blue";
    opts.folder = folder;
    var response =  instance.addSlideNewTextRange(name, slideIndex, shapeIndex, paragraphIndex, opts,
    function (error, data, response) {
      if (error) throw error;
   });
})();